clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -207,9 +207,10 @@ public synchronized void ensureCoordinatorReady() {
     }
 
     protected RequestFuture<Void> lookupCoordinator() {
-        if (findCoordinatorFuture == null) {
-            findCoordinatorFuture = sendGroupCoordinatorRequest();
-            findCoordinatorFuture.addListener(new RequestFutureListener<Void>() {
+        RequestFuture<Void> future = findCoordinatorFuture;
+        if (future == null) {
+            findCoordinatorFuture = future = sendGroupCoordinatorRequest();
```

Comment:
@ijuma Would it really be so bad if occasionally two `sendGroupCoordinatorRequest` instances were created if one was correctly used? I was trying to avoid adding new synchronized blocks - especially since the code looked like it doesn't care, and adding synchronization would require understanding all the locking in the consumer :-). Happy to change to double-checked locking though, if required.